### DISCUSSION TITLE
fatal error: '/lib/modules/6.6.0-rc2-00215-g52bab3a92861/build/include/linux/kconfig.h' file not found

### DISCUSSION THREAD

**Discussion by aaronlu** [bpftrace/bpftrace#2818] [Q&A] [answered]
Hi, I got an error on v6.6 and later kernels when running below scripts:

$ sudo bpftrace -e "tracepoint:kvm:kvm_exit { @exits[args.exit_reason] = count(); }"
fatal error: '/lib/modules/6.6.0-rc2-00215-g52bab3a92861/build/include/linux/kconfig.h' file not found

It appears commit d8131c2965d5e("kbuild: remove $ (MODLIB)/source symlink") has removed the creation of "source" symlink to ksrc and it's merged into v6.6-rc1, so v6.6 and later kernels will not have source symlink under $(MODLIB).

Per my understanding, this affects below setup:
- on v6.6-rc1 and later kernels;
- kobj being different from ksrc(I used a separate directory for build);
- args used in the script.

My current workaround is to manually create the source symlink. I wonder: if there is a legitimate reason to access kconfig.h, perhaps we should let that commit author know that "source" symlink is still needed? Thoughts?

**Comment by viktormalik** [comment, marked as answer]
I think that the problem on your setup is that `/lib/modules/[...]/source` doesn't exist and `/lib/modules/[...]/build` doesn't contain the necessary headers (such as `linux/kconfig.h`). In that case, I'd suggest to use the `BPFTRACE_KERNEL_SOURCE` env variable and pass the path to the kernel source tree.

On a slightly different matter: I believe that we don't need to search for ksrc/kobj at all if we have BTF, which could avoid problems like this. I opened #2884 to track that.

**Comment by aaronlu** [comment]
Thanks viktor and I look forward to that enhancement landing.
